docs/designers-developers/developers/tutorials/block-tutorial/creating-dynamic-blocks.md

```diff
@@ -189,14 +189,14 @@ registerBlockType( 'gutenberg-examples/example-05-dynamic', {
 	icon: 'megaphone',
 	category: 'widgets',
 
-	edit: function( props ) {
+	edit: ( props ) => {
```

Comment:
I would love to see clarification on proper standard for function definition. The coding guidelines here does not include anything: https://developer.wordpress.org/block-editor/contributors/develop/coding-guidelines/

I'm unclear if `edit: function( props ) { }` is preferred to `edit: ( props ) => { }`

If that is the case, then wouldn't it follow that `const varName = function(props) => { }` would be preferred over `const varName = ( props ) => { }` 

A lot of code across Gutenberg uses the shorter fat arrow syntax, which also seems to be the direction of the larger JS community.
whcih

Reply:
I think the difference also exists between ES5 format, and ESNext format.

Reply:
We have this ESLint rule enabled:
https://eslint.org/docs/rules/object-shorthand

See:
https://github.com/WordPress/gutenberg/blob/95e769df1f82f6b0ef587d81af65dd2f48cd1c38/packages/eslint-plugin/configs/esnext.js#L26

Which favours the following:
```js
/*eslint-env es6*/

// properties
var foo = {x, y, z};

// methods
var foo = {
    a() {},
    b() {}
};
```

I guess between a regular function and the fat arrow syntax you can't force one as they have some differences in how they behave:
- regular function is hoisted
- arrow function inherits scope from the parent

In general, I'd love us to find a way to start linting code examples in some way. I don't know how, but it feels like this would make it easier to review all documentation changes :)